### ISSUE TITLE
Bind dirs creation failing

### ISSUE THREAD

**Issue by kototama** [QubesOS/qubes-issues#5862]
**Qubes OS version**
4.0

**Affected component(s) or functionality**
bind-dirs

**Brief summary**
Following instructions at https://www.qubes-os.org/doc/bind-dirs/ does not work.

**To Reproduce**
- I started a debian appvm
- I follow instruction from https://www.qubes-os.org/doc/bind-dirs/ to create a `/nix` directory
- I restart the appvm
- No `/nix` directory is created 

**Expected behavior**
`/nix` should be created upon reboot

**Actual behavior**
Nothing is created.  Running `/usr/lib/qubes/init/bind-dirs.sh` manually shows so errors that `nix` does not exist in `/rw/bind-dirs` (can't copy the logs because I rebooted).

**Solutions you've tried**
Creating `/rw/bind-dirs/nix` manually worked.

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**
_A list of links to the Qubes documentation (or other relevant software documentation) pages you have already consulted._

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues
none

**Comment by marmarek** [comment]
Please show _exactly_ how have you configured it.

**Comment by kototama** [comment]
In the App VM:

```
sudo -i
mkdir -p /rw/config/qubes-bind-dirs.d
echo "binds+=( '/nix' )" > /rw/config/qubes-bind-dirs.d/50_user.conf
poweroff
```

After restart:

```
ls /nix
ls: cannot access '/nix': No such file or directory
```

**Comment by fepitre** [comment]
Create the directory /rw/bind-dirs/nix too.

**Comment by kototama** [comment]
Yes, I figured that but the documentation says:

> Creation of the files and folders in /rw/bind-dirs should be automatic the first time the TemplateBasedVM is restarted after configuration.

**Comment by fepitre** [comment]
Yes something is missing. We can do either a doc fix or a script fix. I let to @marmarek the choice.

**Comment by unman** [comment]
@fepitre  I'm not convinced that anything is missing.
I dont see anything in the doc that suggests that making an entry in the bind-dirs configuration will **create** a file/directory **in the root file system**
The only thing it says is that making an entry there will create files in /rw/bind-dirs, as the quote from @kototama says.

bind-dirs is not used to *create* files in the root filesystem. It's used to make user changes to those existing files persistent. That's clear throughout the doc page, and particularly in the ["How does it work" section](https://www.qubes-os.org/doc/bind-dirs/#how-does-it-work).
This is misuse.
The only guard, beside a warning in the docs, would be to monitor the config files and check that files/directories referred to are existent, but since it's possible to create root files *after* a config here, I'm not sure that is the answer.

**Comment by kototama** [comment]
It would make sense to clarify the documentation then. Specially it's counter-intuitive to create a folder in `/` since from experience the user know it's read-only.

In addition the wording TemplateBasedVM can be confusing. For example the creator of this gist thought that bind-dirs are created in the template:

https://gist.github.com/tomfitzhenry/13fb504f983ba1af34944ac17f27bee4

**Comment by unman** [comment]
There's actually nothing wrong with putting bind-dirs directives in to a template, as should be clear from the docs use of "intermediate priority, for template wide configuration"

**Comment by kototama** [comment]
Ok but once again it's not clear from the sentence `/etc/qubes-bind-dirs.d (intermediate priority, for template wide configuration)` that this should be done in the template since all the documentation before referred to editing file in the app VM.

**Comment by unman** [comment]
True that

**Comment by UndeadDevel** [comment]
It would be nice if we could get the associated PR merged...just ran into this issue and wasted several hours trying to figure out why it didn't work due to outdated docs.

@unman can you please answer deeplow's comment in the PR? This issue has been stuck in that PR for over 3 years and the convo there is now again stuck for over a month...FWIW I also think deeplow is right and unman's objection in the PR convo thread doesn't apply.

Maybe the PR should just be merged to help people trying to figure out how bind-dirs works? Any additional objections can be addressed with a new issue and PR.
